controllers/clustertopology_controller_sync.go

```diff
@@ -16,10 +16,68 @@ limitations under the License.
 
 package controllers
 
-import "context"
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
 
 // reconcileState reconciles the current and desired state of the managed Cluster topology.
 func (r *ClusterTopologyReconciler) reconcileState(ctx context.Context, current, desired *clusterTopologyState) error {
-	// TODO: add reconcile logic
 	return nil
 }
+
+type twoWaysMergePatchHelper struct {
+	client   client.Client
+	original client.Object
+	data     []byte
+}
+
+// createTwoWaysMergePatchHelper will return a patch that yields the modified document when applied to the original document.
+// NOTE: In the case of ClusterTopologyReconciler, original is the current object, modified is the desired object, and
```

Comment:
Should we standardize on either modified or desired and use only one of those?

Reply:
Ah nope, please ignore that one

Reply:
I don't think so; createTwoWaysPatchHelper solves a generic problem and it is a good candidate for being moved to another package.